### DISCUSSION TITLE
Portals permissions

### DISCUSSION THREAD

**Discussion by FunSkelet** [Zrips/CMI#7191] [Enhancement]
Hello, Zrips.

I'm doing a mission on my project where people have a limited travel area. I do these zones through CMI Portals. However, I need these zones to not work when the player is not completing a mission. However, in order to do this, at the beginning and end of the mission, it is necessary to issue and take away the rights cmi.commands.portal.name . But this only puts an extra burden on the plugin that controls the rights.

It would be nice if it were possible to set custom rights to use portals.

For example, the right on the mission.robbery portals, which simultaneously, when issued, grants access to the use of these portals

**Comment by mrfloris** [comment]
you could not set the end point for the portal, but instead run commands. 
You can then use the aliaseditor to make an internal command like `/_checkPortal`that takes and checks for certain params. And set the portal command to this with `asConsole! _checkPortal [playerName] uniqueportalID whatever
then $1 is the playername, $2 is the unique portal id and $3 is whatever.

portal command will be like this for example for the 5th portal:
`asConsole _portalCheck [playerName] 5`

and the custom aliaseditor made command `/_portalCheck` then has this info (I run into it, the username might be mrfloris...) and id 5.

You can then run asConsole! commands and check: and perm: things (specialized commands) to see if the user has a perm, if it's which portal, and what to do then ..  like `check:$2=5! perm:cmi.portalcheck.4! sConsole! lp user $1 permission set cmi.portalcheck.5 true` and eh also `check:$2=5! perm:cmi.portalcheck.4@! msg! $1 You cannot enter this portal, you need to finish zone 4 first`

Using some more complex sum and integer combos you can even automated some of the msgs to calculate current+1 or -1 to show and compare against the right stuff - simplifying your code. 

https://www.zrips.net/cmi/commands/specialized/
https://www.zrips.net/cmi/portals/
https://www.zrips.net/cmi/commands/custom-alias/
